cmd/benchfilesort/main.go

```diff
@@ -80,15 +80,15 @@ func encodeRow(b []byte, row *comparableRow) ([]byte, error) {
 	sc := &stmtctx.StatementContext{TimeZone: time.Local}
 	body, err = codec.EncodeKey(sc, body, row.key...)
 	if err != nil {
-		return b, errors.Trace(err)
+		return b, errors.WithStack(err)
```

Comment:
Why not simply `return b, err`?

Reply:
this is first step of replacement,  finally this case will be `b, err`

(1)replace all errors.Trace to errors.Withstack (2) delete the "middle call" of `errors.WithStack` 

we can not repalce `b, errors.Trace` directly, because same "root call" maybe introduce some standard errors or third-part lib error which is not `pkg error` and need to  do a wrap..

It's hard to use script to detect it's "middle call" or "root call", so replace them to WithStack, then manual delete middle one orz..

or @disksing  is any other good advise~?

Reply:
Reasonable and fair. Good luck.

Reply:
This seems like a deficiency in the design of pkg/errors that we could remedy by just adding a new function that will take a stack trace if it exists? This has been [mentioned by others](https://github.com/pkg/errors/issues/75).

See this error implementation where they coalesce stack traces when printing: https://github.com/upspin/upspin/blob/master/errors/debug.go#L26.
Also see this error package which merges stack traces every time they are being wrapped: 
https://github.com/srvc/fail/blob/master/annotators.go#L50

In my PR for improved error codes I am now searching for the deepest stack trace:
https://github.com/pingcap/pd/pull/1200/files#diff-f10e5c1ee6d26ee1746a0a9188a67d4dR32
That assumes use of pkg/errors, but there is also now srvc/fail and go-stack/stack, all with different definitions of what a stack trace are.

Reply:
thanks~ @gregwebs , it makes sense, I try to implement it again, but use bottom-up way mark error already has stack info~ 3712fc1eb4b24c5a439788c3b5a4ee22375f5193